src/foundry/client/data/documents/token.d.ts

```diff
@@ -53,35 +49,48 @@ declare global {
     get combatant(): InstanceType<ConfiguredDocumentClass<typeof Combatant>> | null;
 
     /**
-     * An indicator for whether or not this Token is currently involved in the active combat encounter.
+     * An indicator for whether this Token is currently involved in the active combat encounter.
      */
     get inCombat(): boolean;
 
     /**
-     * Is the Token currently hidden from player view?
+     * Define a sort order for this TokenDocument.
+     * This controls its rendering order in the PrimaryCanvasGroup relative to siblings at the same elevation.
+     * In the future this will be replaced with a persisted database field for permanent adjustment of token stacking.
+     * In case of ties, Tokens will be sorted above other types of objects.
      */
-    get hidden(): boolean;
+    get sort(): number;
+
+    set sort(value: number);
+
+    /** @internal */
+    #sort: number;
+
+    override prepareBaseData(): void;
 
     /**
-     * @param data    - (default: `{}`, unused)
-     * @param options - (default: `{}`, unused)
+     * Prepare detection modes which are available to the Token.
+     * Ensure that every Token has the basic sight detection mode configured.
+     * @internal
      */
+    protected _prepareDetectionModes(): void;
+
     override clone(
-      data?: Parameters<foundry.documents.BaseToken["clone"]>[0],
-      options?: Parameters<foundry.documents.BaseToken["clone"]>[1]
-    ): TemporaryDocument<this>;
+      data?: Parameters<BaseToken["clone"]>[0],
+      options?: Parameters<BaseToken["clone"]>[1]
+    ): this | Promise<this>;
```

Comment:
Now that I look at it again: Is this really already persisted? If I remember correctly, `TemporaryDocument` basically removes the `id` from a document, reflecting the state when it only exists in memory and is not persisted yet.

Reply:
I tried this signature:
```ts
    override clone(
      data?: Parameters<BaseToken["clone"]>[0],
      options?: Parameters<BaseToken["clone"]>[1]
    ): TemporaryDocument<this> | Promise<TemporaryDocument<this>>;
```

And I get this error:
> Property 'clone' in type 'TokenDocument' is not assignable to the same property in base type 'Document<any, any, null> & CanvasDocumentMixin<Document<any, any, null>>'.
  Type '(data?: AssignmentType<Schema, {}>, options?: ({ save?: boolean; keepId?: boolean; } & DocumentConstructionContext) | undefined) => TemporaryDocument<...> | Promise<...>' is not assignable to type '(data?: AssignmentType<any, {}>, { save, ...context }?: ({ save?: boolean; keepId?: boolean; } & DocumentConstructionContext) | undefined) => this | Promise<...>'.
    Type 'TemporaryDocument<this> | Promise<TemporaryDocument<this>>' is not assignable to type 'this | Promise<this>'.
      Type 'TemporaryDocument<this>' is not assignable to type 'this | Promise<this>'.
        Type 'this | { _source: unknown; }' is not assignable to type 'this | Promise<this>'.
          Type '{ _source: unknown; }' is not assignable to type 'this | Promise<this>'.
            Type 'TokenDocument' is not assignable to type 'this | Promise<this>'.
              Type 'TokenDocument' is not assignable to type 'this'.
                'TokenDocument' is assignable to the constraint of type 'this', but 'this' could be instantiated with a different subtype of constraint 'TokenDocument'.
                  Type 'TemporaryDocument<this>' is not assignable to type 'Promise<this>'.
                    Type 'this | { _source: unknown; }' is not assignable to type 'Promise<this>'.
                      Type 'this' is not assignable to type 'Promise<this>'.
                        Type 'TokenDocument' is missing the following properties from type 'Promise<this>': then, catch, finally, [Symbol.toStringTag]
                          Type 'TokenDocument' is missing the following properties from type 'Promise<this>': then, catch, finally, [Symbol.toStringTag]ts(2416)